shell.nix

```diff
@@ -18,6 +18,7 @@ mkShell {
     perl
     python3
     bazel
+    jdk11
```

Comment:
This is added to avoid [this error](https://github.com/tweag/rules_haskell/runs/1864772907#step:7:380):
```
       ERROR: /home/runner/.cache/bazel/_bazel_runner/c91220d87f26118b19246641d5266d21/external/local_jdk/BUILD.bazel:3:10: in fail_rule rule @local_jdk//:jdk: 
       Traceback (most recent call last):
       	File "/home/runner/.cache/bazel/_bazel_runner/c91220d87f26118b19246641d5266d21/external/bazel_tools/tools/jdk/fail_rule.bzl", line 19, column 13, in _fail_rule_impl
       		fail("%s %s" % (ctx.attr.header, ctx.attr.message))
       Error in fail: Auto-Configuration Error: Cannot find Java binary bin/java in /home/runner/.cache/bazel/_bazel_runner/install/a704d98024e76cfbf273992773847901/embedded_tools/tools/jdk/nosystemjdk; either correct your JAVA_HOME, PATH or specify embedded Java (e.g. --javabase=@bazel_tools//tools/jdk:remote_jdk11)
       ERROR: Analysis of target '//debug/linking_utils:test-ldd' failed; build aborted: Analysis of target '@local_jdk//:jdk' failed
       INFO: Elapsed time: 1.937s
       INFO: 0 processes.
       FAILED: Build did NOT complete successfully (0 packages loaded, 304 targets configured)
       ERROR: Couldn't start the build. Unable to run tests
       FAILED: Build did NOT complete successfully (0 packages loaded, 304 targets configured)
       INFO: Streaming build results to: https://app.buildbuddy.io/invocation/2c151cbc-1caa-483a-835e-ba30de50c5ff
       FAILED: Build did NOT complete successfully (0 packages loaded, 304 targets configured)
```
A better solution would be a rules_nixpkgs provided Java toolchain, but that is out of scope for this PR.